infra/modules/providers/azure/app-service/main.tf

```diff
@@ -6,6 +6,16 @@ locals {
   acr_webhook_name               = "cdhook"
   app_names                      = keys(var.app_service_config)
   app_configs                    = values(var.app_service_config)
+
+  app_linux_fx_versions = [
+    for config in values(var.app_service_config) :
+    // Without specifyin a `linux_fx_version` the webapp created by the `azurerm_app_service` resource
+    // will be a non-container webapp.
+    //
+    // The value of "DOCKER" is a stand-in value that can be used to force the webapp created to be
+    // container compatible without explicitly specifying the image that the app should run.
+    config.image == "" ? "DOCKER" : format("DOCKER|%s/%s", var.docker_registry_server_url, config.image)
```

Comment:
So if no image is provided we set this to docker? Then the test is expecting only "Docker" with no image? How does the end-to-end test with echo server change?

Reply:
No change in E2E. The `"DOCKER"` is simply to force the app service that is created to be a container service (as opposed to other non container types)

Reply:
To elaborate, if you do specify an image (which is what the other templates do) then it will be deployed. hence no change to E2E tests.